[Davian Kho Yong Quan] iP

Daviancold opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Daviancold committed:
Add Davvy.java

Created a skeletal new chatbot Davvy that greets user and exit

Daviancold committed:
Fix typo referring to chatbot name

Daviancold committed:
Add echo feature

A new method that allows user input to be returned back to the command
line was added to help the user ensure the correct message is being
received

Daviancold committed:
Add list feature

A list that allows the saving of user input allows for multiple saved
inputs that can be retrieved later on when requested by the user

Daviancold committed:
Add task marking function

Refactored the code by splitting the adding function of tasks into a
list into a separate class for easier management and future scalability.

Deleted Duke.java as it was useless.

Daviancold committed:
Add new chatbot name variable

Daviancold committed:
Refactor code for future overriding

Main changes in the way user input is parsed, this time taking first
word as command and subsequent word as argument.

Moved printing to helper methods for easier overriding when new classes
are implemented with different printing

Daviancold committed:
Add todo, event and deadline feature

Three new different types of tasks can now be added to the bot, with
new fields for each, such as custom due date and time

wenenhoe commented on src/main/java/Davvy.java:
```diff
@@ -0,0 +1,110 @@
+
+public class Davvy {
+    private static final String BOT_NAME = "Davvy";
+    private static boolean toEnd;
```
Suggest better naming with a more boolean-like name, e.g. `isExitStatus`

Daviancold replied:
Thank you for the suggestion!

wenenhoe commented on src/main/java/Davvy.java:
```diff
@@ -0,0 +1,110 @@
+            printLine();
+            return null;
+        }
+        return inputText.split("\\b", 2);
```
Can include comment to help readers with what `\b` regex is

Daviancold replied:
I will do just that, thanks!

wenenhoe commented on src/main/java/Davvy.java:
```diff
@@ -0,0 +1,110 @@
+            break;
+        case "mark":
+            try {
+                TaskList.getTask(parseInt(commandArg.trim()) - 1).markDone();
```
```suggestion
                int taskIndex = parseInt(commandArg.trim()) - 1;
                TaskList.getTask(taskIndex).markDone();
```

wenenhoe commented on src/main/java/Davvy.java:
```diff
@@ -0,0 +1,110 @@
+            break;
+        case "unmark":
+            try {
+                TaskList.getTask(parseInt(commandArg.trim()) - 1).markNotDone();
```
```suggestion
                int taskIndex = parseInt(commandArg.trim()) - 1;
                TaskList.getTask(taskIndex).markNotDone();
```

wenenhoe commented on src/main/java/Davvy.java:
```diff
@@ -0,0 +1,110 @@
+                Events inputEvent = new Events(newCommandArg[0], newCommandArg2[0], newCommandArg2[1]);
+                TaskList.addTask(inputEvent);
+            } else {
+                System.out.println("Please put a correct time!");
```
Good job on printing error prompt so that the user knows what is the issue

yyangdaa replied:
Good job on printing error prompt

wenenhoe reviewed (commented):
Looks good
Some cosmetic changes suggested

yyangdaa commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,12 @@
+    public String toString() {
+        return "[D]" + super.toString() + "(by:" + dueDate + ")";
+    }
+}
```
Good Job in applying the concept of inheritance!

Daviancold committed:
add new exception handling for incorrect user input

Some refactoring was done to all exception handling

Comments from the last pull request were added as well

Daviancold committed:
Merge branch 'branch-Level-5'

Daviancold committed:
Refactor code into packages

Daviancold committed:
Merge branch 'branch-A-Packages'

StanleyW00 commented on src/main/java/NewExceptions/EmptyArgumentException.java:
```diff
@@ -0,0 +1,4 @@
+package NewExceptions;
+
+public class EmptyArgumentException extends Exception {
+}
```
Try to avoid empty classes. Perhaps, add methods to handle the exception. Same goes for the other exception classes.

StanleyW00 commented on src/main/java/Tasking/Davvy.java:
```diff
@@ -0,0 +1,132 @@
+            TaskList.addTask(inputTodo);
+            break;
+        case "deadline":
+            if (commandArg.contains("/by")) {
+                String[] newCommandArg = commandArg.split("/by", 2);
+                Deadline inputDeadline = new Deadline(newCommandArg[0], newCommandArg[1]);
+                TaskList.addTask(inputDeadline);
+            } else {
+                printLine();
+                System.out.println("Please put a date!");
+                printLine();
+            }
```
Perhaps abstract this part into another method would be better. Same goes for other cases in this method.

StanleyW00 commented on src/main/java/Tasking/Task.java:
```diff
@@ -0,0 +1,34 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+    public void markDone() {
+        isDone = true;
+        Davvy.printLine();
+        System.out.println(" Nice! I've marked this task as done:");
+        System.out.println(this);
+        Davvy.printLine();
+    }
+    public void markNotDone() {
+        isDone = false;
+        Davvy.printLine();
+        System.out.println(" OK, I've marked this task as not done yet:");
+        System.out.println(this);
+        Davvy.printLine();
+    }
```
Perhaps ,add spacing between methods would be better for readability. Same goes for the TaskList class.

StanleyW00 reviewed (commented):
Overall, good job in following the coding standards and maintaining the code quality.  Just need to fix a few mistakes mentioned.

Daviancold committed:
Add new line after methods for readability with some refactoring

Collection (ArrayList) already previously implemented

Daviancold committed:
Add Delete feature

Removed repeated code for printing line before and after each command
as well

Daviancold committed:
Add Autosave feature

Checks if directory and file exists as well, if it does not, one will
be created

Daviancold committed:
Merge branch 'branch-Level-6' into branch-Level-7

# Conflicts:
#	src/main/java/Tasking/Davvy.java
#	src/main/java/Tasking/Task.java
#	src/main/java/Tasking/TaskList.java

Daviancold committed:
Bug fixes after merging

Changes to mark status were not saved in the file previously

Daviancold committed:
Merge branch 'branch-Level-7'

Daviancold committed:
Refactor code into new classes

New classes include the Ui and Parser Class. Also renamed ReadWriteFile
to Storage

Daviancold committed:
Add find item feature

search only works for letter to letter matches

Daviancold committed:
Add Java Doc to methods

Daviancold committed:
Merge pull request #2 from Daviancold/branch-Level-9

Add find item feature

Daviancold committed:
Merge branch 'master' into branch-A-JavaDoc

Daviancold committed:
Merge pull request #1 from Daviancold/branch-A-JavaDoc

Add Java Doc to methods

Daviancold committed:
Update README.md

Made small bug fixes to printing output

Daviancold committed:
Fix typo in README.md
